tests/fixtures.py

```diff
@@ -0,0 +1,156 @@
+import os
+import pytest
+import logging
+import ocs.ocp
+import ocs.defaults as defaults
+from utility import templating
+
+logger = logging.getLogger(__name__)
+
+
+# Ceph Block Pool
+CBP = ocs.ocp.OCP(
+    kind='CephBlockPool', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+PVC = ocs.ocp.OCP(
+    kind='PersistentVolumeClaim', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+SC = ocs.ocp.OCP(
+    kind='StorageClass', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+Pod = ocs.ocp.OCP(
+    kind='Pod', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+OCP = ocs.ocp.OCP(
+    kind='Service', namespace=defaults.ROOK_CLUSTER_NAMESPACE
+)
+
+TEMP_YAML = os.path.join("templates/ocs-deployment", "temp.yaml")
+TEMPLATES_DIR = "templates/ocs-deployment"
+PROJECT_NAME = 'test-project'
+
+
+@pytest.fixture()
+def create_ceph_block_pool():
+    """
+    Create a Ceph block pool
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "CephBlockPool.yaml")
+    logger.info(f'Creating a Ceph Block Pool')
+
+    templating.dump_to_temp_yaml(template, TEMP_YAML)
+    assert CBP.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    # TODO:
+    # wait()
+
+
+@pytest.fixture()
+def create_storageclass():
+    """
+    Create a storage class
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "StorageClass.yaml")
+    logger.info(f'Creating a storage class')
+
+    templating.dump_to_temp_yaml(template, TEMP_YAML)
+    assert SC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    # TODO:
+    # wait()
+
+
+@pytest.fixture()
+def create_pvc():
+    """
+    Create a persistent Volume Claim
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "PersistentVolumeClaim.yaml")
+    logger.info(f'Creating a PVC')
+
+    templating.dump_to_temp_yaml(template, TEMP_YAML)
+
+    assert PVC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    # TODO:
+    # wait()
+
+
+@pytest.fixture()
+def create_pod():
+    """
+    Create a pod
+
+    """
+    template = os.path.join(TEMPLATES_DIR, "Pod.yaml")
+    logger.info(f'Creating a pod')
+
+    templating.dump_to_temp_yaml(template, TEMP_YAML)
+
+    assert PVC.create(yaml_file=TEMP_YAML)
+    open(TEMP_YAML, 'w').close()
+    # TODO:
+    # wait()
+
+
+@pytest.fixture()
+def delete_ceph_block_pool(request):
```

Comment:
why not add this to the finalizer to create ceph block pool?
Same for all finalizers below

Reply:
done